frontend/src/Components/NewCard Body/NewCardForm/CardForm.tsx

```diff
@@ -0,0 +1,227 @@
+import React, { useReducer, useEffect } from "react";
+import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
+import { useNavigate } from "react-router-dom";
+import Card from "@material-ui/core/Card";
+import { Button } from "../../Button/Button";
+import { Container } from "./CardForm.style";
+import * as Yup from "yup";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    card: {
+      marginTop: theme.spacing(10),
+      borderColor: "while",
+      padding: "20px",
+    },
+  })
+);
+
+type State = {
+  cardName: string;
+  cardContent: string;
+  cardImg: string;
+  isButtonDisabled: boolean;
+  helperText: string;
+};
```

Comment:
Change the name of this type to "Card" or something similar.
Bonus- Create a file which contains all the types of your application and import the type from that file

Reply:
Anyway, it would be best if you manage this form using Formik instead of using Redux